core/app/models/spree/store_credit_ledger_entry.rb

```diff
@@ -0,0 +1,49 @@
+# Financial transaction entry for a specific `store_credit`
+module Spree
+  class StoreCreditLedgerEntry < Spree::Base
+    PENDING = 'pending'
+    LIABILITY = 'liability'
+
+    belongs_to :store_credit
+    belongs_to :originator, polymorphic: true
+
+    scope :chronological, -> { order(:created_at) }
+    scope :reverse_chronological, -> { order(created_at: :desc) }
+    scope :liability, -> { where(liability: true) }
+    scope :pending, -> { where(liability: false) }
+
+    delegate :currency, to: :store_credit
+
+    def self.credit(store_credit, amount, originator, liability_type=LIABILITY)
+      create!(
+        {
+          store_credit: store_credit,
+          amount: amount,
+          originator: originator,
+          liability: liability_type == LIABILITY
+        }
+      )
+    end
+
+    def self.debit(store_credit, amount, originator, liability_type=LIABILITY)
+      # make sure debit amounts are stored as negative number
+      amount = -amount if amount > 0
```

Comment:
Yes, that and with documentation on the method. Magic vs Convenience, I will make sure to get the documentation on the method done first. :)